pkg/selectors/kernel.go

```diff
@@ -392,6 +421,16 @@ func writeMatchRangesInMap(k *KernelSelectorState, values []string, ty uint32) e
 			// If only one value in the string, e.g. "5", then add it a second time to simulate
 			// a range that starts and ends with itself, e.g. as if "5:5" had been specified.
 			rangeStr = append(rangeStr, rangeStr[0])
+
+			// Special actions for particular network ops
+			switch op {
+			case SelectorOpProtocol:
+				protocol, err := network.InetProtocolNumber(v)
+				if err == nil {
+					protocolStr := fmt.Sprintf("%d", protocol)
+					rangeStr = []string{protocolStr, protocolStr}
+				}
```

Comment:
Can we return an error here? We do fail silently in other places, but I don't think is a good practice.

Reply:
We don't need to as it gets picked up when trying to convert to a number. This part just tries to convert the text string as a IPPROTO_ string. If it works, great, but if it doesn't then the string falls through and is attempted to be converted to a number in the following loop. So the logic was to try the special case, and if it fails, do the usual case.

Reply:
I see. Thanks!